app/connectors/Acc44Connector.scala

```diff
@@ -0,0 +1,166 @@
+import play.api.http.Status.{BAD_REQUEST, CREATED, INTERNAL_SERVER_ERROR, OK}
+import play.api.libs.json._
+import play.api.{Logger, LoggerLike}
+import services.{DateTimeService, MetricsReporterService}
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import utils.JSONSchemaValidator
+
+import javax.inject.Inject
+import scala.collection.immutable.Seq
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class Acc44Connector @Inject()(httpClient: HttpClient,
+                               appConfig: AppConfig,
+                               dateTimeService: DateTimeService,
+                               jsonSchemaValidator: JSONSchemaValidator,
+                               metricsReporterService: MetricsReporterService,
+                               headers: MdgHeaders)(implicit ec: ExecutionContext) {
+
+  val log: LoggerLike = Logger(this.getClass)
+
+  def cashAccountTransactionSearch(reqDetails: CashAccountTransactionSearchRequestDetails): Future[Either[ErrorDetail,
+    CashAccountTransactionSearchResponseContainer]] = {
+
+    val commonRequest = CashTransactionsRequestCommon(
+      originatingSystem = MDTP,
+      receiptDate = dateTimeService.currentDateTimeAsIso8601,
+      acknowledgementReference = headers.acknowledgementReference)
+
+    val cashAccTransSearchRequest = CashAccountTransactionSearchRequest(commonRequest, reqDetails)
+
+    val cashAccTransSearchRequestContainer: CashAccountTransactionSearchRequestContainer =
+      CashAccountTransactionSearchRequestContainer(cashAccTransSearchRequest)
+
+    jsonSchemaValidator.validatePayload(
+      Json.toJson(cashAccTransSearchRequestContainer), jsonSchemaValidator.acc44RequestSchema) match {
+
+      case Success(_) => postValidRequest(cashAccTransSearchRequestContainer)
+
+      case Failure(exception) =>
+        log.error(s"Request validation failed against the schema and error is ::::: ${exception.getMessage}")
+        Future(Left(
+          ErrorDetail(
+            dateTimeService.currentDateTimeAsIso8601,
+            "MDTP_ID",
+            BAD_REQUEST.toString,
+            exception.getMessage,
+            mdtp,
+            SourceFaultDetail(Seq(REQUEST_SCHEMA_VALIDATION_ERROR)))
+        ))
+    }
+  }
+
+  private def postValidRequest(cashAccTransSearchRequestContainer: CashAccountTransactionSearchRequestContainer):
+  Future[Either[ErrorDetail, CashAccountTransactionSearchResponseContainer]] = {
+
+    metricsReporterService.withResponseTimeLogging("hods.post.cash-account-transaction-search") {
+      httpClient.POST[CashAccountTransactionSearchRequestContainer, HttpResponse](
+        appConfig.acc44CashTransactionSearchEndpoint,
+        cashAccTransSearchRequestContainer,
+        headers = headers.headers(appConfig.acc44BearerToken, None)
+      )(implicitly, implicitly, HeaderCarrier(), implicitly).map { res =>
+        res.status match {
```

Comment:
I think you want this with a space between it and 89 for readability?

Reply:
Done